File: LeetCode/Medium/0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;
        int s = 0;

        for (int i = 0; i < n; i++) {
            s += gas[i] - cost[i];
        }
        
        if (s < 0) {
            return -1;
        }
        
        int idx = 0;
        int curGas = 0;

        for (int i = 0; i < n; i++) {
            if (i > 0) {
                curGas -= cost[i-1];
            }
            if (curGas < 0) {
                idx = i;
                curGas = 0;
            }
            curGas += gas[i];
        }
        return idx;
    }
}

File: LeetCode/Medium/0134-gas-station/0134-gas-station.go
func canCompleteCircuit(gas []int, cost []int) int {
    n := len(gas)

    sumGas := 0
    sumCost := 0
    
    idx := 0
    curGas := 0

    for _, num := range gas {
        sumGas += num
    }

    for _, num := range cost {
        sumCost += num
    }

    if (sumGas < sumCost) {
        return -1
    }

    for i := 0; i < n; i++ {
        if (curGas < 0) {
            curGas = 0
            idx = i
        }
        curGas += gas[i]
        curGas -= cost[i]
    }
    return idx
}